bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java

```diff
@@ -356,10 +356,10 @@ public Object encode(Object msg, ByteBufAllocator allocator) throws Exception {
 
     private static ByteBuf serializeProtobuf(MessageLite msg, ByteBufAllocator allocator) {
         int size = msg.getSerializedSize();
-        ByteBuf buf = allocator.heapBuffer(size, size);
+        ByteBuf buf = allocator.directBuffer(size, size);
```

Comment:
shall we add a comment that explains why this is better ?

Reply:
I will do it. For reviewers: it is the same as one in PR description